### PR TITLE
Hopefully fix crashes(?)

### PR DESCRIPTION
**Pull request by Bleuzen**
I noticed that Carla as a flatpak is more unstable compared to Carla running native. One way to get it to crash over and over again is when playing something from YouTube in Firefox and trying to route this playback through x42-eq stereo. This always makes Carla as flatpak crash, but not non-flatpak Carla.

This is not ready for merge, just testing / guessing, not sure exactly what the problem could be. Remember from the initial PR, I removed the `"--system-talk-name=org.freedesktop.RealtimeKit1"`, but @hfiguiere said:

> Some app crash if they don't have access to it - speaking from experience here.

So I readded it, hope it helps, will test when I have some time.
If it not works, I'm very open to suggestions :)

### CODE REVIEW

**Comment by Bleuzen** [comment]
ew, didn't PRs auto build at some point? or is it because of the branch name?

**Comment by Bleuzen** [comment]
bot, build

**Comment by hfiguiere** [comment]
there seem to be lags in auto detecting. it should automatically build on creation and new commits.

File: studio.kx.carla.json [hunk 1]
```diff
@@ -8,11 +8,12 @@
     "rename-icon": "carla",
     "finish-args": [
         "--share=ipc",
-        "--socket=x11",
         "--socket=wayland",
+        "--socket=fallback-x11",
```
Comment (hfiguiere) [inline studio.kx.carla.json:12]: no, you really want to disable wayland here. Plugins UI won't work if not running with X11.

Comment (Bleuzen) [reply studio.kx.carla.json:12]: ah okay thanks, will change it back

File: studio.kx.carla.json [hunk 2]
```diff
@@ -8,11 +8,12 @@
     "rename-icon": "carla",
     "finish-args": [
         "--share=ipc",
-        "--socket=x11",
         "--socket=wayland",
```
Comment (hfiguiere) [inline studio.kx.carla.json:11]:
```suggestion
```

I really meant to remove wayland too.

**Comment by falkTX** [comment]
the audio goes through pipewire, right?
so that would be the reason for the crashes.

some discussion at https://github.com/falkTX/Carla/issues/1434

still not fully fixed on pipewire's side.

**Comment by Bleuzen** [comment]
@falkTX 
> the audio goes through pipewire, right?

Yes

> so that would be the reason for the crashes.

Could be, was not far away reporting a pipewire bug, but tried with non-flatpak Carla before (also using pipewire) and it doesn't crash with that. Could still have to do something with pipewire tho, don't know yet.

File: studio.kx.carla.json (hunk 2 above)
Comment (Bleuzen) [reply studio.kx.carla.json:11]: this was already added in the initial PR as developer wanted Carla to have wayland access. Is this really problematic / have you tested it?
afaik this doesn't force wayland, only permits.
I just tested this in a wayland session, Carla running as flatpak with wayland access.
x42 plugins, LSP plugins, Surge, Vitalium and guitarix plugins run and display their UI just fine.
In which cases this cause problems?

**Comment by Bleuzen** [comment]
> Build [71711 successful](https://flathub.org/builds/#/builders/32/builds/71711)

still crashes

File: studio.kx.carla.json (hunk 2 above)
Comment (hfiguiere) [reply studio.kx.carla.json:11]: ok then

**Comment by Bleuzen** [comment]
Tested this with latest pipewire git (master @ 0753e992b82adb46c1cc11a34225b6ed9f2c8335) on host, still crashes. Not sure how to change pipewire version in flatpak, guess it is not that easy.
Still not sure if this is even a pipewire bug or not.
Tried debugging, this is the output when running Carla with gdb:
https://pastebin.com/raw/XPx0yGDD

This are the last lines while it runs fine:
```
[New Thread 0x7fffcbfff640 (LWP 488)]
[New Thread 0x7fffcb7fe640 (LWP 489)]
[New Thread 0x7fffcaffd640 (LWP 490)]
Carla assertion failure: "nframes == pData->bufferSize" in file CarlaEngineJack.cpp, line 2889, v1 2048, v2 1024
[New Thread 0x7fffca7fc640 (LWP 491)]
[New Thread 0x7fffc9ffb640 (LWP 492)]
```
this is in idle. Now when I add a plugin (for example x42-eq, but also happens with built in 3bandeq), it crashes and these lines add to the output:

```
Download failed: No route to host.  Continuing without debug info for /app/extensions/Plugins/lv2/fil4.lv2/fil4.so.
Download failed: No route to host.  Continuing without debug info for /usr/lib/x86_64-linux-gnu/libcairo.so.2.
Download failed: No route to host.  Continuing without debug info for /usr/lib/x86_64-linux-gnu/libpixman-1.so.0.
[New Thread 0x7fffc8a9c640 (LWP 496)]
[New Thread 0x7fffbbfff640 (LWP 497)]
[New Thread 0x7fffbb7fe640 (LWP 498)]
[New Thread 0x7fffbaffd640 (LWP 499)]
[New Thread 0x7fffba7fc640 (LWP 500)]
[New Thread 0x7fffb9ffb640 (LWP 501)]

Thread 1 "carla.bin" received signal SIGSEGV, Segmentation fault.
0x00007ffff2bca184 in QWidgetPrivate::getOpaqueChildren() const () from /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
(gdb)
```

Looks not that useful to me.
Noticed there are a lot `Download failed: No route to host.  Continuing without debug info for` messages but couldn't solve this. Read I have to install the debug extensions... this worked for Carla itself:

```
~ ❯ flatpak install --user studio.kx.carla.Debug                                                                                                                                                                                 [134] 3m 28s
Looking for matches…
Skipping: studio.kx.carla.Debug/x86_64/stable is already installed
```

but not for the sdk:

```
~ ❯ LANG=C flatpak install --user org.kde.Sdk.Debug//5.15-21.08
Looking for matches?


        ID                         Branch              Op          Remote          Download
 1. [?] org.kde.Sdk.Debug          5.15-21.08          i           flathub         1.0?kB / 3.7?GB

Error: Failed to read commit 0f7e92348806014602b9f5688b0b32a93123bf6d5bfd94cd382464a078a8805e: No such metadata object 42df5e385888a0ea7c3fab6970ef05be970b528361877548bb82846e02adb6ab.dirtree
error: Failed to install org.kde.Sdk.Debug: Failed to read commit 0f7e92348806014602b9f5688b0b32a93123bf6d5bfd94cd382464a078a8805e: No such metadata object 42df5e385888a0ea7c3fab6970ef05be970b528361877548bb82846e02adb6ab.dirtree
```

Not sure what to do about this. @hfiguiere I'm missing something or is this a bug?

**Comment by Bleuzen** [comment]
As it crashes in `0x00007ffff2bca184 in QWidgetPrivate::getOpaqueChildren() const () from /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5`
maybe we could try another Qt version.
@falkTX  does Carla already support building with (Py)Qt6?

**Comment by falkTX** [comment]
Correct, I havent done any tests regarding Qt6, I am still on Ubuntu 20.04 which does not even have it as option to try.

Regarding the crash, did you try latest carla git? Do you have any carla experimental features enabled?

**Comment by Bleuzen** [comment]
> Correct, I havent done any tests regarding Qt6, I am still on Ubuntu 20.04 which does not even have it as option to try.

Hm would be interesting if Carla can already build with Qt6 if it would search for it.



> Regarding the crash, did you try latest carla git?

not yet but soon, new test build is running



> Do you have any carla experimental features enabled?

The "Enable experimental features" checkbox is off.

---

Found something:
Usually I use the "Engine > Core > Process mode" = "Multiple Clients"
in Carla
and with that it crashes.
But when I use the Process mode "Single Client" instead it doesn't crash.

**Comment by Bleuzen** [comment]
@falkTX  Carla from latest git behaves just the same. Still crashes in "Multiple Clients" mode.

**Pull request closed without merging**